internal/wrappers/application-http.go

```diff
@@ -0,0 +1,54 @@
+package wrappers
+
+import (
+	"encoding/json"
+	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"net/http"
+)
+
+type ApplicationsHTTPWrapper struct {
+	path string
+}
+
+func NewApplicationsHTTPWrapper(path string) ApplicationsWrapper {
+	return &ApplicationsHTTPWrapper{
+		path: path,
+	}
+}
+
+func (a *ApplicationsHTTPWrapper) Get(params map[string]string) (*ApplicationsResponseModel, *ErrorModel, error) {
+	if _, ok := params[limit]; !ok {
+		params[limit] = limitValue
+	}
+
+	clientTimeout := viper.GetUint(commonParams.ClientTimeoutKey)
+
+	resp, err := SendHTTPRequestWithQueryParams(http.MethodGet, a.path, params, nil, clientTimeout)
+	defer resp.Body.Close()
+
+	if err != nil {
+		return nil, nil, err
+	}
+	decoder := json.NewDecoder(resp.Body)
+
+	switch resp.StatusCode {
+	case http.StatusBadRequest, http.StatusInternalServerError:
+		errorModel := ErrorModel{}
+		err = decoder.Decode(&errorModel)
```

Comment:
what are we doing with this error (we don't want the user to see internal erros)? what about logs?

Reply:
From what I see, in cases where it's a recognized failed http code we map it to the errorModel and later mapped to a Golang error. We can come up with a better solution to this and apply to all of the project.

Reply:
We need to make sure we are not returning the user the error, so if we get the error,m let's map to the right message (failed getting X /  bad request to X ) and log the error.

Reply:
Fixed